python/tvm/driver/tvmc/common.py

```diff
@@ -38,6 +41,36 @@
 class TVMCException(Exception):
     """TVMC Exception"""
 
+class TVMCSilentArgumentParser(argparse.ArgumentParser):
```

Comment:
just curious: was there a reason we couldn't use `add_help=False` for each TVMC ArgumentParser, and then manually add it ourselves at the last minute? this approach requires us to use `_remove_action`, which is marked private.

Reply:
@areusch Hi. PTAL at https://github.com/apache/tvm/pull/9229/commits/ffd19e868adb51505843d8090ec61054c6ef1480 

The approach of adding the help options at the last minute works :) Now I think the code is fine. Adding at the last minute for each parser didn't work well, but it was in fact only necessary to do that to the main parser (in main.py) and additionally ignore the `invalid choice` error in the main parser too by overriding the `error` method in `argparse.ArgumentParser`. I've changed the name of `TVMCSilentArgumentParser` to `TVMCSuppressedArgumentParser`, since that class can't suppress all the error, like for instance, the "required argument" error (but that's not necessary).

Reply:
If you're ok now I'll rebase that branch on main and promote it to a PR so the CI can kick in.

Reply:
aewsome, that looks great!